drivers/sqliter-driver/src/commonMain/kotlin/com/squareup/sqldelight/sqliter/driver/SQLiterHelper.kt

```diff
@@ -0,0 +1,228 @@
+package com.squareup.sqldelight.sqliter.driver
+
+import co.touchlab.sqliter.*
+import co.touchlab.stately.collections.frozenHashMap
+import co.touchlab.stately.collections.frozenLinkedList
+import co.touchlab.stately.concurrency.AtomicBoolean
+import co.touchlab.stately.concurrency.AtomicReference
+import co.touchlab.stately.concurrency.QuickLock
+import co.touchlab.stately.concurrency.withLock
+import co.touchlab.stately.freeze
+import com.squareup.sqldelight.Transacter
+import com.squareup.sqldelight.db.*
+import com.squareup.sqldelight.db.SqlPreparedStatement.Type.*
+
+class SQLiterHelper(private val databaseManager: DatabaseManager) : SqlDatabase, EnforceClosed by EnforceClosedImpl() {
+    internal val connectionList = frozenLinkedList<SQLiterConnection>(stableIterator = false)
+
+    override fun close() {
+        checkNotClosed()
+        trackClosed()
+        connectionList.forEach { it.close() }
+        databaseManager.close()
+    }
+
+    override fun getConnection(): SqlDatabaseConnection {
+        checkNotClosed()
+        val conn = SQLiterConnection(databaseManager.createConnection())
+        connectionList.add(conn)
+        return conn
+    }
+}
+
+/**
+ * For simplicity with the current API, the connection can be shared between threads. In theory, however,
+ * in a "Saner Concurrency" context, we'd be better off if connections were tied to threads.
+ *
+ * SQLite cannot nest transactions, so there should be no situation where we have an enclosing transaction.
+ * Transactions are isolated to connections, though, so if you're looking to have multiple transactions,
+ * create multiple connections. This won't help a lot from a performance perspective because only one
+ * connection can write at one time.
+ */
+class SQLiterConnection(
+        internal val databaseConnection: DatabaseConnection
+) : SqlDatabaseConnection, EnforceClosed by EnforceClosedImpl() {
+    private val transaction: AtomicReference<Transaction?> = AtomicReference(null)
+    private val transLock = QuickLock()
+    private val statementList = frozenLinkedList<Statement>(stableIterator = false)
+    private val queryList = frozenLinkedList<SQLiterQuery>(stableIterator = false)
```

Comment:
If the user closes these connections, open statements need to be closed as well for sqlite to be happy.